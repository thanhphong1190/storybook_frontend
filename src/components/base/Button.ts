import styled from 'styled-components'
import tw from 'twin.macro'

export const Button = styled.button`
  // base
  ${tw`px-4 py-4 lg:py-2 transition-colors duration-300 uppercase italic`}
  
  // primary
  ${props =>
    (!!!props.color || props.color == 'orange') &&
    tw`bg-orange hover:bg-orange__dark text-white`}
  
  // secondary
  ${props =>
    props.color == 'blue' && tw`bg-blue hover:bg-blue__dark text-white`}
  
  // white
  ${props =>
    props.color == 'white' && tw`bg-white hover:bg-grey__light text-blue`}
`

export const ButtonLink = styled.a`
  // base
  ${tw`px-4 py-4 lg:py-2 transition-colors duration-300 uppercase italic`}
  
  // primary
  ${props =>
    (!!!props.color || props.color == 'orange') &&
    tw`bg-orange hover:bg-orange__dark text-white`}
  
  // secondary
  ${props =>
    props.color == 'blue' && tw`bg-blue hover:bg-blue__dark text-white`}
  
  // white
  ${props =>
    props.color == 'white' && tw`bg-white hover:bg-grey__light text-blue`}
`

export default Button
