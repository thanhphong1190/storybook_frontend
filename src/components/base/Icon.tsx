import {
  FontAwesomeIcon,
  FontAwesomeIconProps
} from '@fortawesome/react-fontawesome'
import {
  faFacebook,
  faYoutubeSquare,
  faTwitter,
  faInstagramSquare
} from '@fortawesome/free-brands-svg-icons'
import { faStar } from '@fortawesome/free-solid-svg-icons'
export const Icons = {
  facebook: faFacebook,
  twitter: faTwitter,
  instagram: faInstagramSquare,
  youtube: faYoutubeSquare,
  star: faStar
}

export const Icon = (props: FontAwesomeIconProps) => {
  return <FontAwesomeIcon {...props} />
}
export default Icon
