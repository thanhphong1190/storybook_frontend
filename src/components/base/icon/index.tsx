export { default as SVGIcon, IconEnums } from './svgIcon'
