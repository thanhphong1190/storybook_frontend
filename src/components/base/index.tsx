export { Button, ButtonLink } from './Button'
export { SVGIcon, IconEnums } from './icon/svgIcon'
export { Icon, Icons } from './Icon'
