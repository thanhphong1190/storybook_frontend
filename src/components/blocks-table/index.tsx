export { BlocksTable, BlocksTableProps } from './BlocksTable'
