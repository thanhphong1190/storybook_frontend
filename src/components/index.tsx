export { ContentBlock } from './blocks/ContentBlock'
export { Header } from './header/Header'
export { Navigation } from './navigation/Navigation'
export { HeroImage } from './hero/HeroImage'
export { KeyMessage } from './key-message/KeyMessage'
export { Footer } from './footer/Footer'
export { default as Banner } from './banner/Banner'
export { default as OrangeBanner } from './orange-banner/OrangeBanner'
export { ReviewCard, CustomerReviews } from './customer-reviews'
export { StandardContent } from './standard-content'
export { FranchiseBenefitCard, FranchiseBenefits } from './franchise-benefits'
export { Hero } from './hero/Hero'
export { ReadyStartedBlocks } from './ready-started-blocks'
export { BlocksTable } from './blocks-table'
