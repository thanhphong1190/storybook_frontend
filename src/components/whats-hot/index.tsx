export { default as WhatsHot, WhatsHotProps } from './WhatsHot'
