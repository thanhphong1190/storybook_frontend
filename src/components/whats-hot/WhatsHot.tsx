import {
  ArticleCard,
  InstagramCard,
  TwitterCard,
  SocialIconCard,
  SocialCardProps
} from './cards'
import * as S from './WhatsHot.styles'
import * as G from '../../styles/global.styles'
import { ButtonLink } from '../base'
import { Link } from '../../services/api/types'

export interface BlockProps {
  type: 'twitter' | 'article' | 'instagram' | 'social-icons'
  block: SocialCardProps
}

interface BlockRow {
  blocks: [BlockProps, BlockProps, BlockProps]
}

export interface WhatsHotProps {
  title: string
  blocksRow?: BlockRow[]
  action: Link
}

export const WhatsHot = (props: WhatsHotProps) => {
  const {
    title,
    blocksRow,
    action: { label, url }
  } = props
  return (
    <S.Section>
      <S.Container>
        <S.TitleWrapper>
          <S.Title>{title}</S.Title>
          <G.Underline />
        </S.TitleWrapper>
        <S.BlocksGroup>
          {blocksRow &&
            blocksRow.map((row, i) => (
              <S.Blocks key={i}>
                {row &&
                  row.blocks.map((block, j) => {
                    switch (block.type) {
                      case 'article':
                        return <ArticleCard key={j} {...block.block} />
                      case 'instagram':
                        return <InstagramCard key={j} {...block.block} />
                      case 'twitter':
                        return <TwitterCard key={j} {...block.block} />
                      case 'social-icons':
                        return <SocialIconCard key={j} {...block.block} />
                    }
                  })}
              </S.Blocks>
            ))}
        </S.BlocksGroup>
        <S.CenteredButton>
          <ButtonLink href={url} color='blue' className='py-4'>
            {label}
          </ButtonLink>
        </S.CenteredButton>
      </S.Container>
    </S.Section>
  )
}

export default WhatsHot
