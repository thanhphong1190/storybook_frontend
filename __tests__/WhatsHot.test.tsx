import renderer from 'react-test-renderer'
import { WhatsHot, WhatsHotProps } from '../src/components/whats-hot/WhatsHot'
import { IconEnums } from '../src/components/base'

const props: WhatsHotProps = {
  title: `What's hot`,
  blocksRow: [
    {
      blocks: [
        {
          type: 'article',
          block: {
            title: 'In fitness by Jasmine Green',
            content: `3 reason to add strength training`,
            action: { label: 'Read article', url: '#' },
            imgURL: '/example/Social-card-article-bg.png'
          }
        },
        {
          type: 'article',
          block: {
            title: 'In fitness by Jasmine Green',
            content: `3 reason to add strength training`,
            action: { label: 'Read article', url: '#' },
            imgURL: '/example/Social-card-article-bg.png'
          }
        },
        {
          type: 'social-icons',
          block: {
            title: 'Follow Us',
            socials: [
              { icon: IconEnums['facebook-c'], url: '#' },
              { icon: IconEnums['twitter-c'], url: '#' },
              { icon: IconEnums['instagram-c'], url: '#' }
            ]
          }
        }
      ]
    }
  ],
  action: {
    label: 'Access to the blog',
    url: '#'
  }
}

it(`What's hot renders correctly`, () => {
  const tree = renderer.create(<WhatsHot {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
