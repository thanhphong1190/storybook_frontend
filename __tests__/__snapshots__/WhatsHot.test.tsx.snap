// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`What's hot renders correctly 1`] = `
<section
  className="WhatsHotstyles__Section-sc-18tlt1v-0 ekxmZk"
>
  <div
    className="WhatsHotstyles__Container-sc-18tlt1v-1 hkpLAf"
  >
    <div
      className="WhatsHotstyles__TitleWrapper-sc-18tlt1v-2 Rhydw"
    >
      <span
        className="WhatsHotstyles__Title-sc-18tlt1v-3 gVfYdg"
      >
        What's hot
      </span>
      <hr
        className="globalstyles__Underline-sc-1nc69wi-8 dVEiXG"
      />
    </div>
    <div
      className="WhatsHotstyles__BlocksGroup-sc-18tlt1v-4 fvVtYQ"
    >
      <div
        className="WhatsHotstyles__Blocks-sc-18tlt1v-5 lgmXkv"
      >
        <div
          className="Cardsstyles__Wrapper-m8f1ab-0 kRJueY"
          style={
            Object {
              "backgroundImage": "url('/example/Social-card-article-bg.png')",
            }
          }
        >
          <div
            className="Cardsstyles__Container-m8f1ab-1 ckTBHf bg-black bg-opacity-50"
          >
            <div
              className="Cardsstyles__CardContent-m8f1ab-2 iMRiqk"
            >
              <div
                className="Cardsstyles__Title-m8f1ab-5 kGFLhg"
                color="white"
              >
                In fitness by Jasmine Green
              </div>
              <div
                className="Cardsstyles__Title-m8f1ab-5 jtHYdt"
                color="white"
                size="2xl"
              >
                3 reason to add strength training
              </div>
            </div>
            <div
              className="Cardsstyles__CardBottom-m8f1ab-4 gMgdeL"
            >
              <a
                className="Button__ButtonLink-lxhj0a-1 gEZLzW"
                color="orange"
                href="#"
              >
                Read article
              </a>
            </div>
          </div>
        </div>
        <div
          className="Cardsstyles__Wrapper-m8f1ab-0 kRJueY"
          style={
            Object {
              "backgroundImage": "url('/example/Social-card-article-bg.png')",
            }
          }
        >
          <div
            className="Cardsstyles__Container-m8f1ab-1 ckTBHf bg-black bg-opacity-50"
          >
            <div
              className="Cardsstyles__CardContent-m8f1ab-2 iMRiqk"
            >
              <div
                className="Cardsstyles__Title-m8f1ab-5 kGFLhg"
                color="white"
              >
                In fitness by Jasmine Green
              </div>
              <div
                className="Cardsstyles__Title-m8f1ab-5 jtHYdt"
                color="white"
                size="2xl"
              >
                3 reason to add strength training
              </div>
            </div>
            <div
              className="Cardsstyles__CardBottom-m8f1ab-4 gMgdeL"
            >
              <a
                className="Button__ButtonLink-lxhj0a-1 gEZLzW"
                color="orange"
                href="#"
              >
                Read article
              </a>
            </div>
          </div>
        </div>
        <div
          className="Cardsstyles__Wrapper-m8f1ab-0 kRJueY"
        >
          <div
            className="Cardsstyles__Container-m8f1ab-1 iofuKM"
          >
            <div
              className="Cardsstyles__Title-m8f1ab-5 jzatRZ"
              color="white"
              size="4xl"
            >
              Follow Us
            </div>
            <div
              className="Cardsstyles__CardIconsContent-m8f1ab-3 csmCyB"
            >
              <a
                className="Cardsstyles__IconWrapper-m8f1ab-6 cqnHue"
                href="#"
              >
                <span
                  className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 evNslP undefined"
                  clickable={true}
                />
              </a>
              <a
                className="Cardsstyles__IconWrapper-m8f1ab-6 kMZtQA"
                href="#"
              >
                <span
                  className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 evNslP undefined"
                  clickable={true}
                />
              </a>
              <a
                className="Cardsstyles__IconWrapper-m8f1ab-6 cqnHue"
                href="#"
              >
                <span
                  className="Iconstyles__StyledSVGIcon-sc-1cwmlu4-0 evNslP undefined"
                  clickable={true}
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="WhatsHotstyles__CenteredButton-sc-18tlt1v-6 eOGWiF"
    >
      <a
        className="Button__ButtonLink-lxhj0a-1 eZiind py-4"
        color="blue"
        href="#"
      >
        Access to the blog
      </a>
    </div>
  </div>
</section>
`;
